File: NeetCode/02-Stack/C++/stackMain.cpp
//
// Created by Jerry on 4/30/2025.
//
#include <iostream>
#include "Stack.h"

using namespace std;


int main() {
    constexpr int size = 5;
    class Stack stack(size);

    stack.push(1);
    stack.push(2);
    stack.push(3);
    stack.push(4);
    stack.push(5);

    cout << "The size of the stack is currently: " << stack.stackSize() << endl;

    cout << "The element at the top of the stack is: " << stack.peek() << endl;

    for(int i = 0; i < size; ++i) {
        cout << stack.pop() << " ";
    }

    return 0;
}

File: NeetCode/02-Stack/Python/stackMain.py
import Stack

def main():
    myStack = Stack.Stack()
    myStack.push(1)
    myStack.push(2)
    myStack.push(3)
    myStack.push(4)
    myStack.push(5)
    myStack.print()
    print(myStack.pop())
    print(myStack.peek())
    myStack.print()

if __name__ == '__main__':
    main()